aea/cli/fetch.py

```diff
@@ -57,13 +57,38 @@
 def fetch(click_context, public_id, alias, local, remote):
     """Fetch an agent from the registry."""
     ctx = cast(Context, click_context.obj)
+    do_fetch(ctx, public_id, local, remote, alias)
+
+
+def do_fetch(
+    ctx: Context,
+    public_id: PublicId,
+    local: bool,
+    remote: bool,
+    alias: Optional[str] = None,
+    target_dir: Optional[str] = None,
+):
+    """
+    Run the Fetch command.
+
+    :param ctx: the CLI context.
+    :param public_id: the public id.
+    :param local: whether to fetch from local
+    :param remote whether to fetch from remote
+    :param alias: the agent alias.
+    :param alias: the target directory, in case fetching locally.
+    :return: None
+    """
+    enforce(
+        not (local and remote), "'local' and 'remote' options are mutually exclusive."
+    )
     is_mixed = not local and not remote
     ctx.set_config("is_local", local and not remote)
     ctx.set_config("is_mixed", is_mixed)
     if remote:
         fetch_agent(ctx, public_id, alias)
     elif local:
-        fetch_agent_locally(ctx, public_id, alias)
+        fetch_agent_locally(ctx, public_id, alias, target_dir=target_dir)
     else:
         fetch_mixed(ctx, public_id, alias)
```

Comment:
hmm, why do mixed and remote not need a target dir?

Reply:
I think it is just missing.

Reply:
Then we need to add it :). Otherwise the next bug might be waiting here.